Source/inv.cpp

```diff
@@ -1776,39 +1772,37 @@ void CheckInvCut(int pnum, int mx, int my, bool automaticMove)
 						case PC_BARBARIAN:
 							PlaySFX(PS_WARR15, false);
 							break;
-
 						case PC_ROGUE:
 						case PC_BARD:
 							PlaySFX(PS_ROGUE15, false);
 							break;
-
 						case PC_SORCERER:
 							PlaySFX(PS_MAGE15, false);
 							break;
-
 						case PC_MONK:
 							PlaySFX(PS_MONK15, false);
 							break;
+						case NUM_CLASSES:

```

Comment:
This is a bit odd, given this could obviously never happen.

Reply:
The idea of adding an extra enum member to signal the number of elements is a terrible hack. Obviously the static analysis cannot tell that this is one such instance.

It's sad that we have to resort to this... is there something else we can do with C++11 to iterate on the enum without needing this hack @AJenbo ?

Reply:
I don't have the same compiler setup to test, but I'd imagine just using an empty default case should fix the warning.

Reply:
This was the best I could come up with :D

Reply:
To be fair for class SFX code I replaced that with a single function and a global table. So the whole switch case is replaced with `GetPlayerSfx(VOICE_CANTUSE)` and the like.

Reply:
> I don't have the same compiler setup to test, but I'd imagine just using an empty default case should fix the warning.

It would, but I prefer this as it would war anyone adding new classes that they need to add it to this switch.